packages/core/strapi/lib/types/core/attributes/relation.d.ts

```diff
@@ -1,48 +1,62 @@
-import { SchemaUID } from '../../utils';
-import { Attribute, ConfigurableOption, PrivateOption } from './base';
-import { GetAttributesByType, GetAttributesValues } from './utils';
+import type { Attribute, Common, Utils } from '@strapi/strapi';
 
-export type BasicRelationsType = 'oneToOne' | 'oneToMany' | 'manyToOne' | 'manyToMany';
-export type PolymorphicRelationsType = 'morphToOne' | 'morphToMany' | 'morphOne' | 'morphMany';
+export type BasicRelationsType =
+  | 'oneToOne'
+  | 'oneToMany'
+  | 'manyToOne'
+  | 'manyToMany'
+  | 'morphOne'
+  | 'morphMany';
+export type PolymorphicRelationsType = 'morphToOne' | 'morphToMany';
 export type RelationsType = BasicRelationsType | PolymorphicRelationsType;
 
-export interface BasicRelationAttributeProperties<
-  S extends SchemaUID,
-  R extends RelationsType,
-  T extends SchemaUID
-> {
+export type BasicRelationProperties<
+  S extends Common.UID.Schema,
+  R extends BasicRelationsType,
+  T extends Common.UID.Schema
```

Comment:
Since this is the "init" type system PR, we should try to set the foundation for good naming conventions. Unless something is truly generic, I would prefer to have more descriptive generic names, such as TLeft, TRelation, TRight instead of S, R, T.

I would love to hear some more thoughts on this.

(This is just the place I chose to bring this up, but there are lots of places where it's even more relevant than here. Depending on the response to this comment I can track down other places with recommendations.)

Reply:
I see @joshuaellis beat me to this one

Reply:
I like this C# approach, I've answered this in Josh's comment, let's continue the discussion there :slightly_smiling_face: